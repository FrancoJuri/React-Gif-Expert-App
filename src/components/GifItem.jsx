import PropTypes from 'prop-types';

const GifItem = ({ title, url }) => {
    return (
        <div className='card'>
            <a href={url} target='_blank'>
                <img src={url} alt={title} />
            </a>
            <p>{ title }</p>
        </div>
    )
}

GifItem.propTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired,
}

export default GifItem;