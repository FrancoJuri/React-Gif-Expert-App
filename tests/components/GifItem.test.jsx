import { render, screen } from '@testing-library/react'
import GifItem from '../../src/components/GifItem'

describe('GifItem testing', () => {

    const title = 'Valorant';
    const url = 'https://valorant.com/img.jpg';

    test('Should match snapshot', () => {
        const { container } = render(<GifItem title={title} url={url} />);

        expect(container).toMatchSnapshot();
    })

    test('Should display img with the indicated URL and Alt atributte', () => {
        render(<GifItem title={title} url={url} />);
        
        /* expect(screen.getByRole('img').src).toBe(url);
        expect(screen.getByRole('img').alt).toBe(title); */
        const { src, alt } = screen.getByRole('img');
        expect(src).toBe(url);
        expect(alt).toBe(title);
    })

    test('Should display title in the component', () => {
        render(<GifItem title={title} url={url} />);
        expect(title).toBeTruthy();
    })
})