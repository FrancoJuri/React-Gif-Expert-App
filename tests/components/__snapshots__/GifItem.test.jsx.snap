// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifItem testing Should match snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <a
      href="https://valorant.com/img.jpg"
      target="_blank"
    >
      <img
        alt="Valorant"
        src="https://valorant.com/img.jpg"
      />
    </a>
    <p>
      Valorant
    </p>
  </div>
</div>
`;
